src/ring_theory/subring.lean

```diff
@@ -476,6 +478,31 @@ instance : complete_lattice (subring R) :=
 lemma eq_top_iff' (A : subring R) : A = ⊤ ↔ ∀ x : R, x ∈ A :=
 eq_top_iff.trans ⟨λ h m, h $ mem_top m, λ h m _, h m⟩
 
+section
+
+variables (R)
+
+/-- The center of a semiring `R` is the set of elements that commute with everything in `R` -/
+def center : subring R :=
+{ carrier := set.center R,
+  neg_mem' := λ a, set.neg_mem_center,
+  .. subsemiring.center R }
+
+variables {R}
+
+lemma mem_center_iff {z : R} : z ∈ center R ↔ ∀ g, g * z = z * g :=
+iff.rfl
+
+@[simp] lemma center_eq_top (R) [comm_ring R] : center R = ⊤ :=
+set_like.coe_injective (set.center_eq_univ R)
+
+/-- The center is commutative. -/
+instance : comm_ring (center R) :=
+{ ..subsemiring.center.comm_semiring,
+  ..(center R).to_ring}
+
```

Comment:
Should there be simp-lemmas for `subring.center.to_submonoid` and all the other various projections? Should maybe the defs be tagged with `@[simps]` and is that good enough?

Reply:
That's the only projection of interest I think, but yes, that should be there. Added.

I don't think `simps` has been configured to produce nice names for subobjects yet, nor do I think the generated coercion lemma should be `simp`.